pydarn/plotting/rtp.py

```diff
@@ -1037,32 +1043,52 @@ def plot_summary(cls, dmap_data: List[dict],
         # [noise, tfreq, cp, snr, vel, spect, elv]
         # Check if the radar has elevation information if not
         # do not plot elevation
+        scalar_parameters = [('noise.sky', 'noise.search'), ('tfreq', 'nave'),
+                             ('cp')]
+
         try:
             # need to use any because some records at the start
             # can be partial which doesn't mean there is no elv
             # data
-            if any('elv' in d for d in dmap_data) and plot_elv:
-                num_plots = 7
+            if any('elv' in d for d in dmap_data) and plot_elv\
+                    and ('elv' in vector_parameters):
+                num_plots = len(scalar_parameters) + len(vector_parameters)
+            elif 'elv' not in vector_parameters:
+                num_plots = len(scalar_parameters) + len(vector_parameters)
             else:
-                num_plots = 6
+                num_plots = len(scalar_parameters) + len(vector_parameters) - 1
+                vector_parameters.remove('elv')
         except KeyError:
-            num_plots = 6
+            num_plots = len(scalar_parameters) + len(vector_parameters) - 1
+            vector_parameters.remove('elv')
         axes = []
+
+        # Check integrity of vector_parameters before committing to plotting
+        allowed_parameters = ['elv', 'v', 'w_l', 'p_l']
+        if not isinstance(vector_parameters, list):
+            vector_parameters = [vector_parameters]
+        test_list = [i for i in vector_parameters if i not in allowed_parameters]
+        if len(test_list) > 0 :
+            raise Exception('Summary plots allow plotting of the following '
+                            'vector parameters only: v (velocity), '
+                            'p_l (Signal to noise ratio), '
+                            'w_l (spectral width), '
+                            'elv (elevation).')
+
```

Comment:
Would it be good to add a check here to exit out gracefully if `vector_parameters` ends up being empty after the elevation checks?